File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        
        if(root == null) return ans;
        
        ArrayDeque<TreeNode> q = new ArrayDeque();
        
        q.add(root);
        
        while(q.size() > 0) {
            
            TreeNode temp = q.peekLast();
            ans.add(temp.val);
            
            int size = q.size();
            
            while(size --> 0) {
                
                TreeNode curr = q.remove();
                if(curr.left != null)
                    q.add(curr.left);
                
                if(curr.right != null)
                    q.add(curr.right);
                
            }
            
        }
        return ans;
    }
}

File: 199-binary-tree-right-side-view/199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {

        queue<TreeNode*> q;
        q.push(root);
        vector<int> ans;
        
        if(root == nullptr) return ans;
        
        while(q.size() > 0) {
            int size = q.size();
            ans.push_back(q.front()->val);
            while(size --> 0) {
                TreeNode* front = q.front();
                q.pop();
                if(front->right)
                    q.push(front->right);
                if(front->left)
                    q.push(front->left);
            }
        }
        
        return ans;
    }
};